pyramid_oereb/contrib/data_sources/standard/sources/plr.py

```diff
@@ -531,14 +532,30 @@ def read(self, params, real_estate, bbox):  # pylint: disable=W:0221
                     else:
                         # We found spatially related elements. This means we need to extract the actual plr
                         # information related to the found geometries.
+
+                        law_status_of_geometry = []
+                        # get distinct values of law_status for all geometries found
+                        for geometry in geometry_results:
+                            if(geometry.public_law_restriction.law_status not in law_status_of_geometry):
+                                law_status_of_geometry.append(geometry.public_law_restriction.law_status)
+
+                        legend_entries_from_db = []
+                        # get legend_entries per law_status
+                        for law_status in law_status_of_geometry:
+                            legend_entry_with_law_status = [
+                                self.collect_legend_entries_by_bbox(session, bbox, law_status),
+                                law_status
+                            ]
+                            legend_entries_from_db.append(legend_entry_with_law_status)
+
                         self.records = []
-                        legend_entries_from_db = self.collect_legend_entries_by_bbox(session, bbox)
                         for geometry_result in geometry_results:
                             self.records.append(
                                 self.from_db_to_plr_record(
                                     params,
                                     geometry_result.public_law_restriction,
-                                    legend_entries_from_db
+                                    next(elem for elem in legend_entries_from_db
```

Comment:
It looks good in general. I only want to mark that this approach leads to potential decrease of performance because we introduce a query to DB per geometry. Did you check this for a parcel witch big amount of PLR's?

Reply:
The legend_entries used to be loaded once per PLR before as well. 
Now, they are loaded once per law_status, per PLR. So, only if there are more than one law status, the query will be executed more often. But I did not test it explicitly.

The alternative would be to adjust the LegendEntry Record to also store the law_status, I guess.

Reply:
Ok thats true. So from my point of view this is ok.